### PR TITLE
suggestions: add network function

### LINKED ISSUE #506: Idea? - implement __add__ for networks
Hi,

Occasionally we had the situation that there's subnetworks that could be defined separately, maybe we are also interested in solving those subnetwork separatly. Then we want to combine them into one big network.
Would it be an idea to implement the class method __add__ such that network1+network2 makes sense and gives you the combined network? Could make some workflows quite tidy. I guess there's a ton of things to consider there, overlapping buses for instance to have them connected after the merger. So at this stage just an idea.

Best,
Andreas

### PR DESCRIPTION
**Pull request by fneum**
closes #781
closes #506 

@jessLryan many thanks for this very nice first draft! 

I have a couple of suggestions which were easier to express by adding an additional commit to yours and moving the discussion here.

https://github.com/PyPSA/PyPSA/pull/783/commits/eafb7e7b2607e7825bc94a45f5431e40e06f7555

- I renamed the function to `merge` and moved it to `io` module
- you can also use the function with `networkA + networkB` with default settings.
- I added keyword arguments `inplace` and `with_time`
- I enforce that there can be no duplicate indices and identical snapshots (I think this is safer)

I largely did not test the code yet. The test coverage for the different use cases could be extended.

### CODE REVIEW

**Review by jessLryan — COMMENTED** [review]
Why skip these? Could this not be left up to the user to decide?

**Comment by jessLryan** [comment]
I think the method should stay in the components file as it's more closely aligned with the other methods in that file whereas the io file is self-described as containing "Functions for importing and exporting data."

**Comment by jessLryan** [comment]
Thank you for your comments and input :-) I largely agree with the changes and agree that more tests should be written for it in the future

File: pypsa/io.py [hunk 1]
```diff
@@ -1000,6 +1000,68 @@ def import_series_from_dataframe(network, dataframe, cls_name, attr):
...
+        If True, merge into ``network`` in-place, otherwise a copy is made.
+    with_time : bool, default True
+        If False, only static data is merged.
+
+    Returns
+    -------
+    receiving_n : pypsa.Network
+        Merged network, or None if inplace=True
+    """
+    to_skip = {"Network", "SubNetwork"}
```
[the remark this answers is not in the document]
Comment (FabianHofmann) [reply pypsa/io.py]: The subnetworks is a calculated instance that depends on the topology. They always have numbers as names, which are likely overlapping across different networks, so skipping them is the right thing to do.

**Review by FabianHofmann — APPROVED** [review]

**Pull request merged**